### PR TITLE
Freebsd cmake

### PR DESCRIPTION
**Pull request by Fneufneu**
Some cmake fixes to make things works on FreeBSD.
I marked [cmake] commits that can affect other platforms.

### CODE REVIEW

**Comment by hudokkow** [comment]
Looks good.
CMAKE_MAKE_PROGRAM change in https://github.com/xbmc/xbmc/pull/10087/commits/7aad68656c010849ccb65e8f084329edce31d5fb should be in https://github.com/xbmc/xbmc/pull/10087/commits/527cd070af8076fc254c8a12117bbd7742183248

Apart from that, only one other comment: so you're the **one** FreeBSD user we have, heheh.

If you have the time, can you test https://github.com/xbmc/xbmc/pull/9987 commenting https://github.com/xbmc/xbmc/pull/9987/commits/6f5e4a5ee3c34ec00d59312927e11344a50f40c9#diff-2cec9cf9f25dcb68763ba455c63ba01bR324 out?
I'm curious and haven't had the time to test FreeBSD using linux's Install.cmake

File: project/cmake/modules/FindSWIG.cmake [hunk 1]
```diff
@@ -1,6 +1,9 @@
 if(NOT SWIG_EXECUTABLE)
   find_program(SWIG_EXECUTABLE NAMES swig2.0 swig PATH_SUFFIXES swig)
 endif()
+if(NOT SWIG_EXECUTABLE)
+  find_program(SWIG_EXECUTABLE NAMES swig3.0 swig PATH_SUFFIXES swig)
```
Comment (wsnipex) [inline project/cmake/modules/FindSWIG.cmake]: just adding swig3.0 to the NAMES list above should suffice

**Comment by fetzerch** [comment]
Thanks much! LGTM (besides the small minor that wsnipex mentioned already).
If you want you can also mention it in https://github.com/xbmc/xbmc/blob/master/project/cmake/README.md (maybe that way we might even find a second bsd user ;))

**Comment by Fneufneu** [comment]
thanks for the feedback. i will update the pr and look at #9987 next week,
but most of the time, linux cmake files are fine for FreeBSD.

**Comment by Fneufneu** [comment]
PR updated and ready to be merged.
I really need to update the README one day :) and you are more than one user on FreeBSD !
Someone send me a patch to make kodi run on RPi 2 / FreeBSD :)

**Comment by fetzerch** [comment]
Thanks, I have the readme updated together with a few more things. Will push that with my next PR.

File: project/cmake/modules/FindCpluff.cmake [hunk 2]
```diff
@@ -22,7 +22,7 @@ if(NOT WIN32)
                                         --host=${ARCH}
                                         CFLAGS=${defines}
                                         LDFLAGS=${ldflags}
-                      BUILD_COMMAND make V=1)
+                      BUILD_COMMAND ${CMAKE_MAKE_PROGRAM} V=1)
```
Comment (fetzerch) [inline project/cmake/modules/FindCpluff.cmake]: @Fneufneu: I just figured out that this breaks the build if one uses the ninja generator. [CMAKE_MAKE_PROGRAM](https://cmake.org/cmake/help/latest/variable/CMAKE_MAKE_PROGRAM.html) is set to a generator specific tool. But here and also for ffmpeg, we need to use "make". I'll investigate in a fix.

Comment (fetzerch) [inline project/cmake/modules/FindCpluff.cmake]: Fixed in https://github.com/xbmc/xbmc/pull/10130 with 26290aa.

**Pull request merged**